erpnext/crm/doctype/lead/lead.py

```diff
@@ -390,3 +391,26 @@ def set_missing_values(source, target):
 		}
 	}, target_doc, set_missing_values)
 	return target_doc
+
+@frappe.whitelist()
+def filter_territory(doctype, txt, searchfield, start, page_len, filters):
+	"""filter territory"""
+
+	territory_list = get_descendants_of("Territory", filters.get("region"))
+	territory_list.append(filters.get("region"))
+
+	return frappe.get_all('Territory',
+		filters={
+			'parent_territory': ('in', territory_list),
+			'territory_name': ("like", "%{0}%".format(txt))
+		},
+		fields=["name"],
+		as_list=1)
+
+def rearrange_standard_fields():
```

Comment:
I don't think we need this anymore, we should do this directly in the Lead JSON

Reply:
Right now, the priority is moving code cleanly, docfields can be dealt later